### DISCUSSION TITLE
Working with CUE files from other programming languages

### DISCUSSION THREAD

**Discussion by eruiz** [cue-lang/cue#2722] [General]
Hi, I'd like to develop a set of microservices that could work with CUE files. Before start to code, I'd like to know if there are any CUE parser library in some programming language like Typescript, Java, etc.
Thanks

**Comment by verdverm** [comment]
Go is the only current implementation. You can compile to WASM or `.so` files and then use FFI. One thing I do is exec out to the CUE binary from python scripts.

One thing to be aware of, is that the evaluator is not concurrency safe yet, so using it in an API requires being careful, i.e. you need to introduce a mutex into your handlers. If you exec out, you don't have to worry about locks, but it also comes with a lot of OS level overhead

**Comment by eruiz** [comment]
Thanks @verdverm for your quick answer. Are there any safe workaround?

For example, convert to JSON Schema (https://cuetorials.com/first-steps/convert-jsonschema/)  or similar in order to work with the transformed file safely?

By safe I meant to avoid to lose data in the transformation process.

Thanks

**Comment by eruiz** [comment]
An option could be to use a JSON parser to read/write CUE files ... does someone know if there is a way to allow a JSON parser to read CUE files? for example, following a specific way to code CUE files

Thanks
